mln/models/module.py

```diff
@@ -25,14 +25,20 @@ class Module(models.Model):
 	last_harvest_time = models.DateTimeField(default=now)
 	clicks_since_last_harvest = models.PositiveSmallIntegerField(default=0)
 	total_clicks = models.PositiveIntegerField(default=0)
-	is_setup = models.BooleanField(null=True, blank=True)
+	is_setup = models.BooleanField(null=True, blank=True, default=False)
 
 	class Meta:
 		constraints = (models.UniqueConstraint(fields=("owner", "pos_x", "pos_y"), name="module_unique_owner_pos"),)
 
 	def __str__(self):
 		return "%s's %s at pos (%s, %s), %i clicks" % (self.owner, self.item.name, self.pos_x, self.pos_y, self.total_clicks)
 
+	def save(self, *args, **kwargs):
+		""" Check if module isn't set up when it doesn't even need to be. """
+		if not self.is_setupable() and self.is_setup is False: 
+			self.is_setup = None
+		super().save(*args, **kwargs)
+

```

Comment:
The problem is we need to check whether the module needs setup when it is created as well since the module creation function (when a page is saved) doesn't really happen to know or care whether it needs setup. 

`Model.save` runs when a new module is created, so it serves as a migration, validation, and creation handler. Alternatively, I could split it into a migration function and override `Model.create` instead of `Module.save`, and we don't really need validation on every save.